### PR TITLE
CO-3868 Add support language detection support for Letter

### PR DESCRIPTION
**Pull request by TheRaphael0000**
- Use pytesseract as the OCR library
- Use langdetect instead of detectlanguage for language detection (which allows to have a minimal confidence threshold)
- Add the implementation for letters import (+ refactor some qr detection logic)
- Move res.lang.compassion into advanced_translations
- Fix language detection calls
- Fix odoo module dependencies

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,12 @@ addons:
       - postgresql-9.6-postgis-2.3 # travis doesn't know which one to install
       - postgresql-9.6-postgis-2.3-scripts # pre-installed on travis but required for runbot
       - libzbar0  # for pyzbar to work
+      - tesseract-ocr # used by pytesseract
+      - tesseract-ocr-eng
+      - tesseract-ocr-fra
+      - tesseract-ocr-deu
+      - tesseract-ocr-ita
```
Comment (ecino) [inline .travis.yml:22]: I would add portuguese and spanish
```suggestion
      - tesseract-ocr-ita
      - tesseract-ocr-spa
      - tesseract-ocr-por
```

File: advanced_translation/models/langdetect.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class OSD(models.AbstractModel):
+    _name = "langdetect"
+    _description = "Language detection with langdetect"
+
+    default_threshold = 0.85
+    languages_langdetect = ["en", "de", "fr", "it"]
```
Comment (ecino) [inline advanced_translation/models/langdetect.py]:
```suggestion
    languages_langdetect = ["en", "de", "fr", "it", "po", "es"]
```

File: advanced_translation/models/ocr.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import string
+
+from odoo import models
+
+
+class OCR(models.AbstractModel):
+    _name = "ocr"
+    _description = "Optical character recognition with Tesseract"
+
+    _languages_tesseract = ["eng", "fra", "deu", "ita", "osd"]
```
Comment (ecino) [inline advanced_translation/models/ocr.py]:
```suggestion
    _languages_tesseract = ["eng", "fra", "deu", "ita", "por", "spa", "osd"]
```

**Review by ecino — CHANGES_REQUESTED** [review]

File: advanced_translation/__manifest__.py [hunk 4]
```diff
@@ -40,6 +40,7 @@
     "data": [
         "security/ir.model.access.csv",
         "views/advanced_translation_view.xml",
+        "data/res.lang.compassion.csv",
```
Comment (ecino) [inline advanced_translation/__manifest__.py:43]: At line 39, add dependencies```
    "external_dependencies": {"python": ["babel", "pytesseract", "iso639", "langdetect"]},
```

File: advanced_translation/models/langdetect.py (hunk 2 above)
Comment (TheRaphael0000) [reply advanced_translation/models/langdetect.py]:
```suggestion
    languages_langdetect = ["en", "de", "fr", "it", "pt", "es"]
```

**Review by ecino — CHANGES_REQUESTED** [review]
Please fix all the detect_lang calls in the code : 
image

**Review by ecino — APPROVED** [review]

**Pull request merged**